apps/api/src/app/subscribers/usecases/handle-chat-oauth/handle-chat-oauth.usecase.ts

```diff
@@ -0,0 +1,126 @@
+import { Injectable } from '@nestjs/common';
+import axios from 'axios';
+import { CreateSubscriber, CreateSubscriberCommand, decryptCredentials } from '@novu/application-generic';
+
+import { HandleChatOauthCommand } from './handle-chat-oauth.command';
+import {
+  IChannelCredentialsCommand,
+  OAuthHandlerEnum,
+  UpdateSubscriberChannel,
+  UpdateSubscriberChannelCommand,
+} from '../update-subscriber-channel';
+import { ChannelTypeEnum, EnvironmentRepository, IntegrationEntity, IntegrationRepository } from '@novu/dal';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+import { ICredentialsDto } from '@novu/shared';
+
+@Injectable()
+export class HandleChatOauth {
+  readonly SLACK_ACCESS_URL = 'https://slack.com/api/oauth.v2.access';
+  readonly SCRIPT_CLOSE_TAB = '<script>window.close();</script>';
+
+  constructor(
+    private updateSubscriberChannelUsecase: UpdateSubscriberChannel,
+    private integrationRepository: IntegrationRepository,
+    private environmentRepository: EnvironmentRepository,
+    private createSubscriberUsecase: CreateSubscriber
+  ) {}
+
+  async execute(command: HandleChatOauthCommand) {
+    const { credentials: integrationCredentials } = await this.getIntegration(command);
+
+    const webhookUrl = await this.getWebhook(command, integrationCredentials);
+
+    const environment = await this.getEnvironment(command);
+
+    await this.createSubscriber(environment, command, webhookUrl);
+
+    const redirect = integrationCredentials.redirectUrl != null && integrationCredentials.redirectUrl != '';
+
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    return { redirect, action: redirect ? integrationCredentials.redirectUrl! : this.SCRIPT_CLOSE_TAB };
+  }
+
+  private async createSubscriber(environment, command: HandleChatOauthCommand, webhookUrl) {
+    await this.createSubscriberUsecase.execute(
+      CreateSubscriberCommand.create({
+        organizationId: environment._organizationId,
+        environmentId: command.environmentId,
+        subscriberId: command?.subscriberId,
+      })
+    );
+
+    const subscriberCredentials: IChannelCredentialsCommand = { webhookUrl: webhookUrl, channel: command.providerId };
+
+    await this.updateSubscriberChannelUsecase.execute(
+      UpdateSubscriberChannelCommand.create({
+        organizationId: environment._organizationId,
+        environmentId: command.environmentId,
+        subscriberId: command.subscriberId,
+        providerId: command.providerId,
+        credentials: subscriberCredentials,
+        oauthHandler: OAuthHandlerEnum.NOVU,
+      })
+    );
+  }
+
+  private async getEnvironment(command: HandleChatOauthCommand) {
+    const environment = await this.environmentRepository.findById(command.environmentId);
+
+    if (environment == null) {
+      throw new ApiException(`Environment ID: ${command.environmentId} not found`);
```

Comment:
```suggestion
      throw new NotFoundException(`Environment ID: ${command.environmentId} not found`);
```
Though maybe we could make the SubscribersModule dependent of the EnvironmentModule and use here the `GetEnvironment` use case instead of reimplementing it.